src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -182,12 +195,13 @@ public void CleanupGeneratedFiles()
 
                 // Only cleanup for providers that have actually generated a file. This keeps us from
                 // accidentally loading lazy providers on cleanup that weren't used
-                foreach (var provider in _tempFileToProviderMap.Values.Distinct())
+                foreach (var provider in _providers)
                 {
-                    provider.CleanupGeneratedFiles(_workspace);
-                }
+                    if (!provider.IsValueCreated)
+                        continue;
```

Comment:
Perhaps not in this spot, but in `TryAddDocument...`, which is called from an RTD event for every document that is opened, I think its best to avoid the part creation.